js/src/edit-block/components/editor.js

```diff
@@ -40,23 +43,37 @@ import { BrowserURL, Header, Main, Side } from './';
  * @return {React.ReactElement} The editor.
  */
 const Editor = ( { initialEdits, onError, postId, postType, settings } ) => {
-	const [ selectedFieldName, setSelectedFieldName ] = useState( '' );
+	const { block, changeBlockName } = useBlock();
+	const [ selectedField, setSelectedField ] = useState( null );
+	const [ currentLocation, setCurrentLocation ] = useState( DEFAULT_LOCATION );
+	const [ panelDisplaying, setPanelDisplaying ] = useState( BLOCK_PANEL );
```

Comment:
Eventually, maybe `useReducer()` instead of 3 `useState` calls.

Reply:
Hm, maybe not:

>When it's just an independent element of state you're managing: useState

https://kentcdodds.com/blog/should-i-usestate-or-usereducer